osquery/database/database.cpp

```diff
@@ -46,6 +46,7 @@ const std::string kEvents = "events";
 const std::string kCarves = "carves";
 const std::string kLogs = "logs";
 const std::string kDistributedQueries = "distributed";
+const std::string kDistributedRunningQueries = "distributed_running";
```

Comment:
Is there a reason why we're creating another column family for this when we already have the `distributed` one?

As far as we've used them, I see them as high level categories. I know that RocksDB has also some specific behavior around how it handles their SST and some implications on performance, but what I'm also partially worried about is that this also creates again a non-backward compatible database.

Reply:
> Is there a reason why we're creating another column family for this when we already have the distributed one?

I wanted to implement this as an additional+simplest feature/code and not modify existing working code.
If we wanted to reuse `"distributed"` domain, then I would need to modify the following for example:
https://github.com/osquery/osquery/blob/ed1a31a4465623c01aa4c6df475a4642d0717d70/osquery/distributed/distributed.cpp#L88-L92
(I would need to differentiate "queryName -> query" from "query -> timestamp" key/values.)

> but what I'm also partially worried about is that this also creates again a non-backward compatible database.

Why is adding new domains a non-backwards compatible change?

Reply:
I see what you mean; it would require changing the prefix of the keys and then write a migration step, as we have done in the past, to fixup the key names, and it would unfortunately be non-backward compatible anyway.
I should've created a prefix for those keys when I moved it to the new domain ^^'

Anyway, adding column families (the domain) is non-backward compatible because older versions of osquery won't attempt to open that new column family and RocksDB will error out, due to the fact that not all column families have been opened.

Reply:
Oh... I see, that makes sense, thanks for the clarification.

- Curious what the plan forward is:
  - Is supporting rollback of osquery version important for users? (am guessing that there were issues when e.g. carves were introduced to osquery, which introduced a new column family)
  - Should osquery document adding new domains for new features is not supported or discouraged unless necessary?

(Maybe there are docs around this and I missed them, sorry.)

---

A non-breaking alternative is to use the "default" column family, but would require careful use of prefixes throughout the application. It also depends on the feature; for this particular feature the key space will not grow considerably (1 entry per deny-listed query).

Reply:
I think it's fine to add the new domain for now, but I think this definitely raises the question on what we should do going forward.
I think it's mostly an annoyance because the user maybe wants to test a new version, finds a problem, but then it can't go back, and if not careful then it can lose the data (logs and events) that were stored.

I'm not sure but maybe it's possible to just list the all the column families in the DB and use that list to open the DB, so that we don't have to explicitly give them and the unused ones are ignored.

Reply:
Draft PR with this idea: #7712.